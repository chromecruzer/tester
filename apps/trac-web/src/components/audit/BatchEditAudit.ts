import {AuditLocationRecord, AuditMatch, dateNow, formatTracDate, UpdateAuditMatch} from "@trac/datatypes";
import _ from "lodash";
import {BatchEdit} from "../BatchEditA";

export default class BatchEditAudit extends BatchEdit {
  private account: AuditLocationRecord | undefined;
  private author: string | undefined;

  constructor() {
    super();
    this.uuids = [] as string[];
  }

  setAuthor(author) {
    this.author = author;
  }

  setAccount(account) {
    this.account = account;
  }

  matchChange(audit_match: AuditMatch): UpdateAuditMatch {
    return {
      items: this.uuids,
      match: audit_match,
      user: this.author || 'unknown',
    }
  }

  moved(moved_uuid, moved_code, warehouse): UpdateAuditMatch {
    return {
      items: this.uuids,
      match: 'Found In Other Location',
      user: this.author || 'unknown',
      uuid: moved_uuid,
      customerId: moved_code,
      warehouse,
    }
  }

  createItemNote(content, annotate_type) {
    console.log(`create note content for account`, this.account)
    return _.map(this.uuids, s => ({
      date_created: formatTracDate(dateNow()),
      author: this.author,
      annotate_type,
      annotated_uuid: s,
      audit_uuid: this.account?.uuid,
      content
    }))
  }
}
