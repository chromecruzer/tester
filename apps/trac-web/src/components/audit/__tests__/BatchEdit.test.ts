import BatchEditAudit from "../BatchEditAudit";
import {DateTime} from "luxon";

describe('BatchEdit', () => {
  let rowsState, batchEdit, location;
  beforeEach(() => {
    rowsState = {
      getSelectedUuids: jest.fn().mockReturnValue(['uuid1', 'uuid2', 'uuid3']),
    }
    location = {
      uuid: 'location uuid',
      received_date: '07/21/2022',
      location_code: 'location code',
      name: 'name of location',
      scan_date: '06/23/2022',
      status: 'Open',
      auditor: 'Uncle Fester',
      scanner: 'Aunt Jemima'
    }
    jest.spyOn(DateTime, 'now').mockImplementation(() => DateTime.fromObject(
      {year: 2022, month: 7, day: 11}))
    batchEdit = new BatchEditAudit(location, 'I am a writer');
  });
  describe('setSelectedRowsState', () => {
    it('should set the selected rows state object', () => {
      batchEdit.setSelectedRowsState(rowsState);
      expect(batchEdit.selected).toEqual(rowsState);
    });
  });
  describe('matchChange', () => {
    it('should return a match change object with the selected uuids', () => {
      batchEdit.setSelectedRowsState(rowsState);
      const actual = batchEdit.matchChange('Billed');
      expect(actual).toMatchSnapshot();
    });
  });
  describe('moved', () => {
    it('should return a match change object with moved location information', () => {
      batchEdit.setSelectedRowsState(rowsState);
      const actual = batchEdit.moved('moved uuid', 'moved code');
      expect(actual).toMatchSnapshot();
    });
  });
  describe('createNote', () => {
    it('should create a note object with the correct type and content', () => {
      batchEdit.setSelectedRowsState(rowsState);
      const actual = batchEdit.createItemNote('Here is a comment\n with a return', 'Audit');
      expect(actual).toMatchSnapshot();
    });
  });
});
