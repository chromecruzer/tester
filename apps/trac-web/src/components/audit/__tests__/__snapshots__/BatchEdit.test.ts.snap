// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatchEdit createNote should create a note object with the correct type and content 1`] = `
Array [
  Object {
    "annotate_type": "Audit",
    "annotated_uuid": "uuid1",
    "audit_uuid": "location uuid",
    "author": "I am a writer",
    "content": "Here is a comment
 with a return",
    "date_created": "07/11/2022",
  },
  Object {
    "annotate_type": "Audit",
    "annotated_uuid": "uuid2",
    "audit_uuid": "location uuid",
    "author": "I am a writer",
    "content": "Here is a comment
 with a return",
    "date_created": "07/11/2022",
  },
  Object {
    "annotate_type": "Audit",
    "annotated_uuid": "uuid3",
    "audit_uuid": "location uuid",
    "author": "I am a writer",
    "content": "Here is a comment
 with a return",
    "date_created": "07/11/2022",
  },
]
`;

exports[`BatchEdit matchChange should return a match change object with the selected uuids 1`] = `
Object {
  "items": Array [
    "uuid1",
    "uuid2",
    "uuid3",
  ],
  "match": "Billed",
  "user": "I am a writer",
}
`;

exports[`BatchEdit moved should return a match change object with moved location information 1`] = `
Object {
  "customerId": "moved code",
  "items": Array [
    "uuid1",
    "uuid2",
    "uuid3",
  ],
  "match": "Found In Other Location",
  "user": "I am a writer",
  "uuid": "moved uuid",
}
`;
